src/generators/dockerHelper.js

```diff
@@ -169,10 +223,161 @@ function buildCakeshopService(config) {
+    expose:
+      - "8000"
+      - "8088"
+    healthcheck:
+      test: ['CMD', 'curl', '-f', 'http://localhost:8000']
+      interval: 5s
+      timeout: 5s
+      retries: 20
+    ports:
+      - "${config.network.splunkPort}:8000"
+      - "8088:8088"
+      - "8125:8125"
+    volumes:
+      - splunk-var:/opt/splunk/var
+      - splunk-etc:/opt/splunk/etc
+      - ./out/config/splunk/splunk-config.yml:/tmp/defaults/default.yml
+      - ./out/config/splunk/dashboards:/opt/splunk/etc/apps/search/local/data/ui/views
+      - ./out/config/splunk/airport_geo.csv:/opt/splunk/etc/apps/splunk-app-quorum/lookups/airport_geo.csv
+    networks:
+      ${networkName}-net:
+        ipv4_address: ${config.network.splunkIp}`
+}
+
+function buildCadvisorService(config) {
+  const networkName = config.network.name
+  return `
+  cadvisor:
+    image: google/cadvisor:latest
+    container_name: cadvisor
+    hostname: cadvisor
+    command:
+      - --storage_driver=statsd
+      - --storage_driver_host=${config.network.splunkIp}:8125
+      - --docker_only=true
+    user: root
+    volumes:
+      - /:/rootfs:ro
+      - /var/run:/var/run:ro
+      - /sys:/sys:ro
+      - /var/lib/docker/:/var/lib/docker:ro
+    networks:
+      - ${networkName}-net
+    logging: *default-logging`
+}
+
+function buildEthloggerService(config) {
+  const networkName = config.network.name
+  let ethloggers = ''
+
+  config.nodes.forEach((node, i) => {
+    const instance = i+1
+    ethloggers += `
+  ethlogger${instance}:
+    image: splunkdlt/ethlogger:latest
+    container_name: ethlogger${instance}
+    hostname: ethlogger${instance}
+    environment:
+      - ETH_RPC_URL=http://node${instance}:${config.containerPorts.quorum.rpcPort}
+      - NETWORK_NAME=quorum
+      - START_AT_BLOCK=genesis
+      - SPLUNK_HEC_URL=https://${config.network.splunkIp}:8088
+      - SPLUNK_HEC_TOKEN=11111111-1111-1111-1111-1111111111113
+      - SPLUNK_EVENTS_INDEX=ethereum
+      - SPLUNK_METRICS_INDEX=metrics
+      - SPLUNK_INTERNAL_INDEX=metrics
+      - SPLUNK_HEC_REJECT_INVALID_CERTS=false
+      - ABI_DIR=/app/abis
+      - COLLECT_PEER_INFO=true
+      - DEBUG=ethlogger:abi:*
+    depends_on:
+      - node${instance}
+    restart: unless-stopped
+    volumes:
+      - ./out/config/splunk/abis:/app/abis:ro
+      - ethlogger-state${instance}:/app
+    networks:
+      - ${networkName}-net
+    logging: *default-logging`
+  })
+  return ethloggers
+}
+
+function buildTxGenService(hasSplunk, config, pubkeys) {
+  const networkName = config.network.name
+  const splunkLogging = hasSplunk
+    ? `logging: *default-logging` : ``
+  let nodeVars = ''
+  config.nodes.forEach((node, i) => {
+    nodeVars += `
+      - NODE${i + 1}=${node.quorum.ip}:${config.containerPorts.quorum.rpcPort}`
+  });
+  let pubkeyVars = ''
+  pubkeys.forEach((pubkey, i) => {
+    pubkeyVars += `
+      - PUBKEY${i + 1}=${pubkey}`
+  });
+
+  return `
+  tx-gen:
```

Comment:
Change this to just 'txgen' so that it matches the docker container name. Otherwise if you run a command like exec with docker you use 'txgen', but with docker-compose you need to use 'tx-gen'

Reply:
this was removed